### PR TITLE
Add Stage 5: Theme engine

### PR DESCRIPTION
**Pull request by ivan-magda**
## Summary
- Parse VS Code JSON/JSONC color themes (both legacy `settings` and modern `tokenColors` formats)
- Resolve TextMate scope stacks to visual styles (foreground color, background color, font style)
- Support theme merging (base + overlay, e.g. dark_vs + dark_plus = Dark+)
- Prefix-based scope matching with dot boundaries and parent scope constraints

## New files
| File | Description |
|------|-------------|
| `RawTheme.kt` | Internal Gson data classes for theme JSON deserialization |
| `Theme.kt` | `FontStyle` enum, `ResolvedStyle`, `Theme` class with `match(scopes)` |
| `ThemeReader.kt` | `ThemeReader` object — JSONC parsing, hex color conversion, theme construction |
| `ThemeReaderTest.kt` | 13 tests: theme loading, default colors, color/fontStyle parsing |
| `ThemeTest.kt` | 17 tests: scope matching against Dark+ with verified color values |
| `dark_vs.json` / `light_vs.json` | VS Code base theme test resources |

## Test plan
- [x] All 30 new theme tests pass
- [x] All 200 tests pass (no regressions against stages 1–4)
- [x] `./gradlew :core:test` — BUILD SUCCESSFUL

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: core/src/main/kotlin/dev/textmate/theme/ThemeReader.kt [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                    i++ // skip past '/'
+                }
+                else -> sb.append(c)
+            }
+        }
+        i++
+    }
+    return sb.toString().replace(Regex(",\\s*([\\]\\}])"), "$1")
+}
+
```
[the remark this answers is not in the document]
Comment (ivan-magda) [reply core/src/main/kotlin/dev/textmate/theme/ThemeReader.kt]: Switched to clean JSON theme files from vscode-textmate test-cases (matching the TypeScript reference which uses `JSON.parse()`). `stripJsonc()` is deleted entirely — no function, no regex, no edge case.

File: core/src/main/kotlin/dev/textmate/theme/Theme.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        var background: Long? = null
+        var fontStyle: Set<FontStyle>? = null
+
+        for (rule in rules) {
+            if (!matchesScope(leafScope, rule.scope)) continue
+            if (rule.parentScopes != null && !matchesParentScopes(parentStack, rule.parentScopes)) continue
+
+            if (rule.foreground != null) foreground = rule.foreground
+            if (rule.background != null) background = rule.background
+            if (rule.fontStyle != null) fontStyle = rule.fontStyle
```
[the remark this answers is not in the document]
Comment (ivan-magda) [reply core/src/main/kotlin/dev/textmate/theme/Theme.kt]: VS Code themes have <100 tokenColors rules (Dark+ merged has ~70). Linear scan at this size is not a bottleneck. This is a PoC — reverse iteration with early exit or pre-indexing is YAGNI.

File: core/src/test/kotlin/dev/textmate/theme/ThemeTest.kt [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+    private lateinit var darkPlus: Theme
+
+    @Before
+    fun setUp() {
+        val darkVsStream = javaClass.classLoader.getResourceAsStream("themes/dark_vs.json")
+            ?: throw IllegalArgumentException("dark_vs.json not found")
+        val darkPlusStream = javaClass.classLoader.getResourceAsStream("themes/dark_plus.json")
+            ?: throw IllegalArgumentException("dark_plus.json not found")
+        darkPlus = darkVsStream.use { vs ->
```
[the remark this answers is not in the document]
Comment (ivan-magda) [reply core/src/test/kotlin/dev/textmate/theme/ThemeTest.kt]: Same — test code with classpath resources that always exist.
